server/s3_bucket_metadata_v1.cc

```diff
@@ -553,7 +589,9 @@ void S3BucketMetadataV1::remove_global_bucket_account_id_info() {
 
 void S3BucketMetadataV1::remove_global_bucket_account_id_info_successful() {
   s3_log(S3_LOG_INFO, stripped_request_id, "%s Entry\n", __func__);
-  this->handler_on_success();
+
+  this->callback(state = S3BucketMetadataState::missing);
```

Comment:
Split to separate statements, this is misleading.

Reply:
Ok